### ISSUE TITLE
Cloudwatch Alarm doesn't generate a ARN

### ISSUE THREAD

**Issue by kazeshini178** [localstack/localstack#3130]
# Type of request: This is a ...

[*] bug report
[ ] feature request

# Detailed description

When creating a CloudWatch Alarm a resource ARN isnt generated for the Alarm. This causes terraform to throw errors when validating the resources was successfully created for output references.

## Expected behavior

Generate an ARN that can be used to reference the resource with localstack, similar to other localstack resources.

## Actual behavior

```bash
Error: error listing tags for CloudWatch Metric Alarm (): InvalidParameter: 1 validation error(s) found.
- minimum field size of 1, ListTagsForResourceInput.ResourceARN.
```

# Steps to reproduce

## Command used to start LocalStack
```bash
docker run -p 4567:4566 localstack/localstack:latest
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
There are some variables that would need to be replaced.
```
resource "aws_cloudwatch_metric_alarm" "dlq_queue_alarm" {
  alarm_name          = "alarm"
  alarm_description   = ""
  comparison_operator = "GreaterThanOrEqualToThreshold"
  evaluation_periods  = 1
  metric_name         = "ApproximateNumberOfMessagesVisible"
  namespace           = "AWS/SQS"
  period              = 60
  statistic           = "Sum"
  threshold           = 1
  datapoints_to_alarm = 1

  dimensions = {
    "QueueName" = aws_sqs_queue.dlq_for_main_queue.name
  }

  alarm_actions = [
    var.sns_alert_topic
  ]

  tags = {
    Name = "queue-alarm"
  }

  lifecycle {
    ignore_changes = [
      tags
    ]
  }
}
```

**Comment by usmangani1** [comment]
@kazeshini178 

Thanks for reporting the issue.
Can you please help us with the full terraform script using which we can reproduce the issue ? We are trying to reproduce the issue but the terraform script seems incomplete.Can you please help us here so that we can resolve the issue sooner?
Thanks

**Comment by kazeshini178** [comment]
@usmangani1, this should be the minimum terraform needed to replicate the issue.

```
resource "aws_sqs_queue" "main_queue" {
  name                        = "my_queue" 
  redrive_policy = jsonencode({
    deadLetterTargetArn = aws_sqs_queue.dlq_for_main_queue.arn
    maxReceiveCount     = 3
  })
}

resource "aws_sqs_queue" "dlq_for_main_queue" {
  name                        = "my_dlq_queue"
}

resource "aws_sns_topic" "sns_alert_topic" {
  name         = "my-alert-topic"
  display_name = "DLQ Alert"
}

resource "aws_cloudwatch_metric_alarm" "dlq_queue_alarm" {
  alarm_name          = "my-alarm"
  alarm_description   = "Alert that there are records that have failed to be processed"
  comparison_operator = "GreaterThanOrEqualToThreshold"
  evaluation_periods  = 1
  metric_name         = "ApproximateNumberOfMessagesVisible"
  namespace           = "AWS/SQS"
  period              = 60
  statistic           = "Sum"
  threshold           = 1
  datapoints_to_alarm = 1

  dimensions = {
    "QueueName" = aws_sqs_queue.dlq_for_main_queue.name
  }

  alarm_actions = [
    aws_sns_topic.sns_alert_topic
  ] 
}
```

**Comment by whummer** [comment]
Thanks for reporting @kazeshini178 . This should be fixed upstream in spulec/moto#3409 and is now available in LocalStack - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by kazeshini178** [comment]
@whummer  Its seems the `latest`  get stuck in the same loop as the previous bug i logged previously #3062 though the error is different. This seems to prevent the confirmation that the resource has been created
```
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 2020/11/09 10:36:22 [DEBUG] [aws-sdk-go] DEBUG: Retrying Request monitoring/ListTagsForResource, attempt 
14
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 2020/11/09 10:36:22 [DEBUG] [aws-sdk-go] DEBUG: Request monitoring/ListTagsForResource Details:
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: ---[ REQUEST POST-SIGN ]-----------------------------
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: POST / HTTP/1.1
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Host: localhost:4567
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: User-Agent: aws-sdk-go/1.35.2 (go1.14.5; windows; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.13.5 (+https://www.terraform.io)
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Content-Length: 156
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Authorization: AWS4-HMAC-SHA256 Credential=mock_access_key/20201109/eu-west-1/monitoring/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=281d704ee4dad902a6c731acbe06de0ace2620d194d71fb964c0d7d467a74f68
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: X-Amz-Date: 20201109T083622Z
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Accept-Encoding: gzip
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Action=ListTagsForResource&ResourceARN=arn%3Aaws%3Acloudwatch%3Aus-east-1%3A000000000000%3Aalarm%3Alocalstack-employee-update-queue-alarm&Version=2010-08-01
2020-11-09T10:36:22.477+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: -----------------------------------------------------
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 2020/11/09 10:36:22 [DEBUG] [aws-sdk-go] DEBUG: Response monitoring/ListTagsForResource Details:
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: ---[ RESPONSE ]--------------------------------------
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: HTTP/1.1 500 
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Connection: close
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Content-Length: 290
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Access-Control-Allow-Headers: authorization,content-type,content-md5,cache-control,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent,x-amz-target,x-amz-acl,x-amz-version-id,x-localstack-target,x-amz-tagging
2020-11-09T10:36:22.498+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Access-Control-Allow-Origin: *
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Access-Control-Expose-Headers: x-amz-version-id
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Content-Type: text/html; charset=utf-8
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Date: Mon, 09 Nov 2020 08:36:22 GMT
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: Server: hypercorn-h11
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe:
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe:
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: -----------------------------------------------------
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 2020/11/09 10:36:22 [DEBUG] [aws-sdk-go] <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
.0_x5.exe: <title>500 Internal Server Error</title>
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: <h1>Internal Server Error</h1>
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: <p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 2020/11/09 10:36:22 [DEBUG] [aws-sdk-go] DEBUG: Validate Response monitoring/ListTagsForResource failed, 
attempt 14/25, error SerializationError: failed to unmarshal error message
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe:                                                                                                         status code: 500, request id:
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: caused by: UnmarshalError: failed to unmarshal error message
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe:                                                                                                         00000000  3c 21 44 4f 43 54 59 50  45 20 48 54 4d 4c 20 50  |<!DOCTYPE HTML P|
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000010  55 42 4c 49 43 20 22 2d  2f 2f 57 33 43 2f 2f 44  |UBLIC "-//W3C//D|
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000020  54 44 20 48 54 4d 4c 20  33 2e 32 20 46 69 6e 61  |TD HTML 3.2 Fina|
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000030  6c 2f 2f 45 4e 22 3e 0a  3c 74 69 74 6c 65 3e 35  |l//EN">.<title>5|
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000040  30 30 20 49 6e 74 65 72  6e 61 6c 20 53 65 72 76  |00 Internal Serv|
2020-11-09T10:36:22.499+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000050  65 72 20 45 72 72 6f 72  3c 2f 74 69 74 6c 65 3e  |er Error</title>|
 |.<h1>Internal Se|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000070  72 76 65 72 20 45 72 72  6f 72 3c 2f 68 31 3e 0a  |rver Error</h1>.|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000080  3c 70 3e 54 68 65 20 73  65 72 76 65 72 20 65 6e  |<p>The server en|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000090  63 6f 75 6e 74 65 72 65  64 20 61 6e 20 69 6e 74  |countered an int|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 000000a0  65 72 6e 61 6c 20 65 72  72 6f 72 20 61 6e 64 20  |ernal error and |
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 000000b0  77 61 73 20 75 6e 61 62  6c 65 20 74 6f 20 63 6f  |was unable to co|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 000000c0  6d 70 6c 65 74 65 20 79  6f 75 72 20 72 65 71 75  |mplete your requ|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 000000d0  65 73 74 2e 20 45 69 74  68 65 72 20 74 68 65 20  |est. Either the |
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 000000e0  73 65 72 76 65 72 20 69  73 20 6f 76 65 72 6c 6f  |server is overlo|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 000000f0  61 64 65 64 20 6f 72 20  74 68 65 72 65 20 69 73  |aded or there is|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000100  20 61 6e 20 65 72 72 6f  72 20 69 6e 20 74 68 65  | an error in the|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000110  20 61 70 70 6c 69 63 61  74 69 6f 6e 2e 3c 2f 70  | application.</p|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe: 00000120  3e 0a                                             |>.|
2020-11-09T10:36:22.652+0200 [DEBUG] plugin.terraform-provider-aws_v3.10.0_x5.exe:
G] plugin.terraform-provider-aws_v3.10.0_x5.exe: caused by: unknown error response tag, {{ title} []}
```
